cirq-google/cirq_google/calibration/workflow.py

```diff
@@ -826,9 +826,9 @@ def run_calibrations(
 
     if isinstance(sampler, Engine):
         engine: Optional[Engine] = sampler
-    elif isinstance(sampler, QuantumEngineSampler):
-        engine = sampler.engine
-        (processor_id,) = sampler._processor_ids
+    elif isinstance(sampler, ProcessorSampler):
+        processor_id = getattr(sampler.processor, 'processor_id', None)
+        engine = sampler.processor.engine() if processor_id is not None else None
```

Comment:
Yeah, there were some typing shenanigans that made this necessary.  I refactored so that the function below takes a processor instead of an engine, which removes the need for this.